autorest/codegen/__init__.py

```diff
@@ -289,6 +304,12 @@ def _build_code_model_options(self) -> Dict[str, Any]:
             "low_level_client": low_level_client,
             "combine_operation_files": self._autorestapi.get_boolean_value("combine-operation-files", version_tolerant),
             "python3_only": python3_only,
+            "package_mode": self._autorestapi.get_value("package-mode"),
```

Comment:
Some of these are already in the options dict / can be get other ways. For example. client_name, credential_scopes, output_folder, and input_file

Reply:
All these are used to render template. Let me check more if already get it.

Reply:
@msyyc, we already have informnation for all of the flags you've added here, besides package-pprint-name, this is doubling code for no real reason. If you go through the code here you can see how we're already storing this information on the code model (not necessarily just in options)